### ISSUE TITLE
Favorites from Sonos not showing

### ISSUE THREAD

**Issue by mhoogenbosch** [johanfrick/custom-sonos-card#257]
Using latest version of both HA core and Sonos-card (respectively 2023.12.3 and 6.12.4)

With following code on my Windows 11 device, it shows only Spotify and Sonos, thus not showing Reolink, My media and Frigate. This is what I'd like it to have.
```
type: custom:sonos-card
sections:
  - player
  - media browser
widthPercentage: 100
mediaArtworkOverrides:
  - mediaTitleEquals: TV
    imageUrl: https://cdn-icons-png.flaticon.com/512/716/716429.png
    sizePercentage: 40
mediaBrowserTitlesToIgnore:
  - Reolink
  - My media
  - Frigate
topFavorites:
  - 360 Dance
  - Techno Bunker
mediaBrowserItemsPerRow: 1
showVolumeUpAndDownButtons: true
mediaBrowserShowTitleForThumbnailIcons: true
numberOfFavoritesToShow: 5
```
Screenshot of Windows 11:

![image](https://github.com/johanfrick/custom-sonos-card/assets/36899598/970aae23-0865-43f6-95e7-71c7db9a3b9a)

On both my Android 12 and iOS 17.2.1 device, it doesn't seem to be working:

![Afbeelding van WhatsApp op 2023-12-27 om 12 04 33_a44b98af](https://github.com/johanfrick/custom-sonos-card/assets/36899598/75a3adfb-a973-40a3-9dc6-25a9f1bac161)

**Comment by johanfrick** [comment]
It actually seems like you are using different versions of the card on the different devices. Try clearing the cache. Also, please use the latest version of the card (6.14.3) and try with that.
I'm also a bit confused since you've posted 2 issues on github, and they use different yaml config. Is that actually the case?

**Comment by johanfrick** [comment]
mediaBrowserTitlesToIgnore only works in the favourites, and not in the media browser pop up.
I should probably rename it to better explain what it does.

**Comment by mhoogenbosch** [comment]
> mediaBrowserTitlesToIgnore only works in the favourites, and not in the media browser pop up. I should probably rename it to better explain what it does.

Sure would explain a lot. But how do you get the preferred items to favorites, I can't seem to find that.

**Comment by johanfrick** [comment]
https://support.sonos.com/en-ca/article/add-favorites-to-my-sonos

**Comment by mhoogenbosch** [comment]
Besides that; browsing media completely slows down my entire app. Would this be just the media browser? Because I can't seem to reproduce it by browsing trough media.

**Comment by johanfrick** [comment]
That sentence did not make sense to me fully. Could you elaborate?

**Comment by mhoogenbosch** [comment]
> https://support.sonos.com/en-ca/article/add-favorites-to-my-sonos

But that's the sonos favorites. I'll try and record it.

**Comment by johanfrick** [comment]
What do you mean? This card is for Sonos

**Comment by mhoogenbosch** [comment]
> What do you mean? This card is for Sonos

Yes, as I understand it is for Sonos devices, but Spotify can be played too. Or am I wrong and is the only way to work with sonos favorites?

When I hit the 'star' i'm getting navigated to 'all favorites'. This is always empty. When I hit 'browse through media', it shows the media browser with all different resources.

**Comment by johanfrick** [comment]
Go to the official Sonos app. Make sure you have Spotify working in that app. Now in the Sonos app add for instance a Spotify playlist to My Sonos. It will now show up under all favourites in the card.

**Comment by mhoogenbosch** [comment]
https://github.com/johanfrick/custom-sonos-card/assets/36899598/53a328be-284d-4acd-b73f-74cd9cb060b0

I have added a screen recording. As you can see Spotify is connected in the Sonos app. But the all fav stays empty.

**Comment by johanfrick** [comment]
That recording didn't show the star section in the Sonos app

**Comment by mhoogenbosch** [comment]
> That recording didn't show the star section in the Sonos app

you mean this section?

![IMG_8388](https://github.com/johanfrick/custom-sonos-card/assets/36899598/5d55165a-87e5-4d4a-8d34-86b237ca313a)

**Comment by johanfrick** [comment]
Ok, yes, there is obviously something wrong here. Those under "afspeelllijsten" should show under "all favorites".
Do you get any errors in the developer web console in the browser?

**Comment by mhoogenbosch** [comment]
> Ok, yes, there is obviously something wrong here. Those under "afspeelllijsten" should show under "all favorites". Do you get any errors in the developer web console in the browser?

Right, this wasn't clear to me. "All favorites" is equal to the favorites inside the Sonos app. I'll check the Developer logs.

**Comment by mhoogenbosch** [comment]
When I hit the star button, lot's and lots of errors run by.

You need to instruct me to gather something valuable This is what runs by:

```
radio:1 
 Uncaught (in promise) 
{code: 'unknown_error', message: 'Unknown error'}
code
: 
"unknown_error"
message
: 
"Unknown error"
[[Prototype]]
: 
Object
constructor
: 
ƒ Object()
hasOwnProperty
: 
ƒ hasOwnProperty()
isPrototypeOf
: 
ƒ isPrototypeOf()
propertyIsEnumerable
: 
ƒ propertyIsEnumerable()
toLocaleString
: 
ƒ toLocaleString()
toString
: 
ƒ toString()
valueOf
: 
ƒ valueOf()
__defineGetter__
: 
ƒ __defineGetter__()
__defineSetter__
: 
ƒ __defineSetter__()
__lookupGetter__
: 
ƒ __lookupGetter__()
__lookupSetter__
: 
ƒ __lookupSetter__()
__proto__
: 
(...)
get __proto__
: 
ƒ __proto__()
set __proto__
: 
ƒ __proto__()
```
![image](https://github.com/johanfrick/custom-sonos-card/assets/36899598/9d0d09d2-f3a2-4f6a-b2f5-1b80ae1e0085)

**Comment by johanfrick** [comment]
Please install 6.14.4, it is easier to debug. Then try to click into that error above. Clicking the black arrow, all the way to where it leads you to a line of code.

**Comment by mhoogenbosch** [comment]
> Please install 6.14.4, it is easier to debug. Then try to click into that error above. Clicking the black arrow, all the way to where it leads you to a line of code.

Already had .4 installed to validate it wasn't a bug in the .3 version.

I tried drilling down:

![image](https://github.com/johanfrick/custom-sonos-card/assets/36899598/cb9ee8f1-0393-498d-832b-99bda801f6a2)

**Comment by johanfrick** [comment]
@mhoogenbosch Could you try this to see if it helps? Just temporarily. If it is working, I'll see if I can start ignoring non-sonos speakers in the card.

**Comment by johanfrick** [comment]
@mhoogenbosch  I've created a beta release that should hopefully work better, even if you have a player that does not provide favorites (which means you don't need to disable that player)
https://github.com/johanfrick/custom-sonos-card/releases/tag/v6.14.6
Can you please try it out?

**Comment by mhoogenbosch** [comment]
There they are:

![image](https://github.com/johanfrick/custom-sonos-card/assets/36899598/77be12dc-84a9-4df8-bce1-61fc6d85bbef)

**Comment by johanfrick** [comment]
> There they are:
> 
> ![image](https://private-user-images.githubusercontent.com/36899598/293359804-77be12dc-84a9-4df8-bce1-61fc6d85bbef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM4NTA2MjgsIm5iZiI6MTcwMzg1MDMyOCwicGF0aCI6Ii8zNjg5OTU5OC8yOTMzNTk4MDQtNzdiZTEyZGMtODRhOS00ZGY4LWJjZTEtNjFmYzZkODViYmVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyMzEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI5VDExNDUyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5ZDk2NzgyOGM2YjRlOTY0OTNjMTczNWU2YmI0ZGVlNzgzNjMwMTlhYmU2NTAwOTQyNWVhYTg2YWI5ODA4NjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4W14n3ZfrDaYszWw26sPLZJ-jXW4lAkCD0XXGF14t6o)

Great! was it the new version that fixed it, or you disabling the player?

**Comment by mhoogenbosch** [comment]
Only installed the latest .6 release. It feels slow so will try to disable the other players too. Unfortunately not at home right now to test a bit better.

**Comment by johanfrick** [comment]
Can't you simply just not include them in the entities list for the card?

**Comment by mhoogenbosch** [comment]
Reverting back to .4 and removing the above-mentioned code makes the favorites disappear again. Adding the lines brings the favorites back.

Would it be possible to just not enumerate the squeezebox players?

**Comment by mhoogenbosch** [comment]
I'm closing this issue. The problem is with my squeezebox players. Excluding them, or more accurate; only including my Sonos devices solves the issues.

**Comment by johanfrick** [comment]
Great that you've got it working! I'm still going to look into making it more stable. It shouldn't crash just because there are non-Sonos players in home assistant.

**Comment by mhoogenbosch** [comment]
Understandable. Good luck!

**Comment by johanfrick** [comment]
Now only Sonos players are shown: https://github.com/johanfrick/custom-sonos-card/releases/tag/v6.14.7

**Comment by johanfrick** [comment]
> excludeItemsInEntitiesList: false makes it like I would like to. Sorry about this!
@mhoogenbosch That bug has been fixed now with [v6.15.3](https://github.com/johanfrick/custom-sonos-card/releases/tag/v6.15.3). It now defaults to false.
